clang/lib/Driver/Driver.cpp

```diff
@@ -2002,6 +2002,44 @@ void Driver::PrintVersion(const Compilation &C, raw_ostream &OS) const {
   // Print out the install directory.
   OS << "InstalledDir: " << Dir << '\n';
 
+  // Print out build configuration options that impact the compiler's runtime
+  // behavior. Intended for identifying the source of issues when reproducing
+  // changes.
+  std::vector<std::string> BuildOptions = {
+#if !__OPTIMIZE__
+      "+unoptimized",
```

Comment:
Question for reviewers: Is this a good enough mechanism for the "unoptimized" build case, or do we actually want to plumb the `CMAKE_BUILD_TYPE` through?

Reply:
Building clang/LLVM with MSVC is a supported configuration and is tested with post-commit bots, so that is definitely an expected configuration. I'm not exactly sure what the best course of action is though. I don't think there's a universally supported preprocessor macro for whether or not optimizations are enabled.

`CMAKE_BUILD_TYPE`  I think would be the best bet. https://github.com/llvm/llvm-project/pull/72021 made it so that there are a fixed number of values (unless additional flags are set). That would also let the user differentiate between something like `RELEASE` which is `-O3` and `MINSIZEREL`, which I believe is `-Oz`. If there is a different value specified, just displaying something like custom I think should be fine.

I don't really have any opinions on this though and will leave to others to decide what the best course of action should be. Just trying to provide additional context.

Reply:
I think `+unoptimized` is good enough for me. `CMAKE_BUILD_TYPE` doesn't give as much signal as it suggests, since toolchain files can stick in whatever CXX flags they want per build type.

Reply:
I was also thinking about the fact that the build type doesn't directly correspond to optimized or not.
I should at least make it so it at least doesn't always display `+unoptimized` on MSVC though, I can fix that.

Reply:
It turns out `llvm/Support/CommandLine` has a definition that's approximately correct for MSVC and uses `__OPTIMIZE__` for gnu compatible compilers, I switched to using that.